catalog/harvest/catalog-harvest-metacardadapter/src/main/java/org/codice/ddf/catalog/harvest/adapter/MetacardOnlyAdapter.java

```diff
@@ -0,0 +1,202 @@
+      throw new HarvestException(
+          String.format("Unable to transform resource [%s]", resource.getName()));
+    }
+
+    UpdateRequest updateRequest = new UpdateRequestImpl(updateId, metacard);
+
+    runWithSubjectOrDefault(
+        () -> {
+          try {
+            UpdateResponse response;
+            response = catalogFramework.update(updateRequest);
+
+            List<String> updatedMetacardIds =
+                response
+                    .getUpdatedMetacards()
+                    .stream()
+                    .map(update -> update.getNewMetacard().getId())
+                    .collect(Collectors.toList());
+
+            if (updatedMetacardIds.size() == 1) {
+              return updatedMetacardIds.get(0);
+            } else {
+              LOGGER.debug(
+                  "Received {} metacards for a single updated resource. [{}]",
+                  updatedMetacardIds.size(),
+                  updatedMetacardIds);
+              throw new HarvestException(
+                  String.format(
+                      "Multiple metacards returned for single update for resource [%s]",
+                      resource.getName()));
+            }
+          } catch (IngestException | SourceUnavailableException e) {
+            LOGGER.debug("Failed to update resource [{}].", resource.getName(), e);
+            throw new HarvestException(
+                String.format("Failed to update resource [%s]", resource.getName()), e);
+          }
+        });
+  }
+
+  @Override
+  public void delete(String id) {
+    DeleteRequest deleteRequest = new DeleteRequestImpl(id);
+
+    runWithSubjectOrDefault(
+        () -> {
+          try {
+            DeleteResponse response = catalogFramework.delete(deleteRequest);
+
+            if (response.getDeletedMetacards().isEmpty()) {
+              LOGGER.debug("No metacards retrieved from catalog delete request for id [{}].", id);
+              throw new HarvestException(
+                  String.format(
+                      "No metacards retrieved from catalog delete request for id [%s]", id));
+            }
+          } catch (IngestException | SourceUnavailableException e) {
+            LOGGER.debug("Failed to delete harvested resource with id [{}].", id, e);
+            throw new HarvestException(
+                String.format("Failed to delete harvested resource with id [%s]", id), e);
+          }
+
+          return null;
+        });
+  }
+
+  private <T> T runWithSubjectOrDefault(final Callable<T> callable) {
+    return SECURITY.runAsAdmin(
+        () -> {
+          try {
+            return SECURITY.runWithSubjectOrElevate(callable);
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.debug("Error executing code with subject", e);
+            return null;
+          }
+        });
+  }
+
+  private Map<String, Serializable> getSecurityMap() {
+    Subject subject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
```

Comment:
❓ Could we use Guava's `ImmutableMap.of(SecurityConstants.SECURITY_SUBJECT, SECURITY.runAsAdmin(SECURITY::getSystemSubject))`?

Reply:
I think I tried this but there were modification exceptions in other parts of the system that were modifying the request arguments map.